### PR TITLE
Rename lakehouse monitor to quality monitor

### PR DESCRIPTION
**Pull request by arpitjasa-db**
## Changes

Update the `databricks_lakehouse_monitor` resource to `databricks_quality_monitor` to reflect naming changes in Go SDK.

## Tests

- [X] `make test` run locally
- [X] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [X] using Go SDK

### CODE REVIEW

File: catalog/resource_lakehouse_monitor.go [hunk 1]
```diff
@@ -12,11 +12,11 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-const lakehouseMonitorDefaultProvisionTimeout = 15 * time.Minute
+const qualityMonitorDefaultProvisionTimeout = 15 * time.Minute
 
 func WaitForMonitor(w *databricks.WorkspaceClient, ctx context.Context, monitorName string) error {
-	return retry.RetryContext(ctx, lakehouseMonitorDefaultProvisionTimeout, func() *retry.RetryError {
-		endpoint, err := w.LakehouseMonitors.GetByTableName(ctx, monitorName)
+	return retry.RetryContext(ctx, qualityMonitorDefaultProvisionTimeout, func() *retry.RetryError {
```
Comment (arpitjasa-db) [inline catalog/resource_lakehouse_monitor.go:18]: NOTE: the Go SDK changes will go live with the next Go SDK release (currently scheduled for this Friday)

**Review by xxxyyy3600 — APPROVED** [review]
I am not very familiar with terraform code but renaming LGTM

**Review by alexott — CHANGES_REQUESTED** [review]
We have non-zero number of customer used this resource in the last 4 weeks.
We need to decide how to proceed with this change - introduce a new resource for example, or break after a communication.

**Comment by arpitjasa-db** [comment]
> We have non-zero number of customer used this resource in the last 4 weeks. We need to decide how to proceed with this change - introduce a new resource for example, or break after a communication.

@alexott @xxxyyy3600 @mgyucht deprecated the old resource and created a new one instead. LMK if this works

**Comment by alexott** [comment]
I converted to Draft until we get new SDK merged

File: catalog/resource_online_table.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ import (
 const onlineTableDefaultProvisionTimeout = 45 * time.Minute
 
 func waitForOnlineTableCreation(w *databricks.WorkspaceClient, ctx context.Context, onlineTableName string) error {
-	return retry.RetryContext(ctx, lakehouseMonitorDefaultProvisionTimeout, func() *retry.RetryError {
+	return retry.RetryContext(ctx, qualityMonitorDefaultProvisionTimeout, func() *retry.RetryError {
```
Comment (nkvuong) [inline catalog/resource_online_table.go:20]: looks like a copy paste error

```suggestion
	return retry.RetryContext(ctx, onlineTableDefaultProvisionTimeout, func() *retry.RetryError {
```

File: catalog/resource_online_table.go [hunk 3]
```diff
@@ -40,7 +40,7 @@ func waitForOnlineTableCreation(w *databricks.WorkspaceClient, ctx context.Conte
 }
 
 func waitForOnlineTableDeletion(w *databricks.WorkspaceClient, ctx context.Context, onlineTableName string) error {
-	return retry.RetryContext(ctx, lakehouseMonitorDefaultProvisionTimeout, func() *retry.RetryError {
+	return retry.RetryContext(ctx, qualityMonitorDefaultProvisionTimeout, func() *retry.RetryError {
```
Comment (nkvuong) [inline catalog/resource_online_table.go:43]:
```suggestion
	return retry.RetryContext(ctx, onlineTableDefaultProvisionTimeout, func() *retry.RetryError {
```

**Comment by arpitjasa-db** [comment]
> I converted to Draft until we get new SDK merged

Thanks @alexott, I saw the SDK hasn't been merged yet, what's the timeline on this? cc @mgyucht

**Review by alexott — APPROVED** [review]
code looks good, waiting for integration tests result

**Comment by alexott** [comment]
Integration tests passed

**Review by hectorcast-db — APPROVED** [review]

**Pull request merged**